src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableValueStore.kt

```diff
@@ -114,6 +117,17 @@ class VariableValueStore(
         .fetchOne(DSL.max(VARIABLE_VALUES.ID))
   }
 
+  /** Returns a project's highest value ID, by variable * */
+  fun fetchMaxValueId(projectId: ProjectId, variableId: VariableId): VariableValueId? {
+    return dslContext
+        .select(DSL.max(VARIABLE_VALUES.ID))
+        .from(VARIABLE_VALUES)
+        .where(VARIABLE_VALUES.PROJECT_ID.eq(projectId))
+        .and(VARIABLE_VALUES.VARIABLE_ID.eq(variableId))
+        .and(VARIABLE_VALUES.IS_DELETED.eq(false))
```

Comment:
Should deleting a value count as a modification that triggers a notification? I think this filter will effectively ignore delete operations but maybe that's what we want.

Reply:
Removed for now. Likely one to look at with product.